app/models/user.rb

```diff
@@ -91,6 +111,17 @@ def email_changed?
     authorizations.blank?
   end
 
+  def build_identity_group
+    raise StandardError, "Identity Group Exists" if identity_group
```

Comment:
There really shoudn't be a way for build_identity_group to be invoked twice, so I didn't think it needed a custom error class. I will add a spec for it though.